### PR TITLE
Skills crud

### PR DESCRIPTION
**Pull request by abror9109**

### CODE REVIEW

File: app/MoonShine/Resources/SkillResource.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        return [
+            FiltersAction::make(trans('moonshine::ui.filters')),
+        ];
+    }
+
+    private static function MakeHierarchicalOptions(&$skills = [], $parent_id = null, $level = 0)
+    {
+        $skill_items = Skill::query();
+
```
Comment (Lanser0614) [inline app/MoonShine/Resources/SkillResource.php:54]: relation bilan oganiz yaxshi

Comment (Lanser0614) [reply app/MoonShine/Resources/SkillResource.php:54]: Skill::query();
Skill::query()->with('children');

File: app/MoonShine/Resources/SkillResource.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public static string $title = 'Skills';
+
+    public function fields(): array
+    {
+        return [
+            ID::make()->sortable(),
+            Text::make('Name'),
+            Text::make('Icon'),
+            Select::make('Parent skill')->nullable()->options($this->GetSkillOptions())->searchable()
```
Comment (Lanser0614) [inline app/MoonShine/Resources/SkillResource.php:25]: Select::make('Parent skill')->nullable()->options($this->GetSkillOptions())->searchable()
Select::make('Parent skill')->nullable()->default(callback yozas);

File: app/MoonShine/Resources/SkillResource.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public function actions(): array
+    {
+        return [
+            FiltersAction::make(trans('moonshine::ui.filters')),
+        ];
+    }
+
+    private static function MakeHierarchicalOptions(&$skills = [], $parent_id = null, $level = 0)
```
Comment (Lanser0614) [inline app/MoonShine/Resources/SkillResource.php]: bunaqa narsala keremas static funksiyala

File: app/MoonShine/Resources/SkillResource.php (hunk 1 above)
Comment (abror9109) [reply app/MoonShine/Resources/SkillResource.php:54]: Hozir bu logikada children ishlatilmayapti, relation bilan olish manimcha keremas

File: app/MoonShine/Resources/SkillResource.php (hunk 2 above)
Comment (abror9109) [reply app/MoonShine/Resources/SkillResource.php:25]: Callback hozir nimaga kerak tushunmadim, select ishlatilgani sababi tushunarliro bolishi uchun ierarxiya bilan korsatish uchun

File: app/MoonShine/Resources/SkillResource.php (hunk 3 above)
Comment (abror9109) [reply app/MoonShine/Resources/SkillResource.php]: Private method ga ozgaritirldi

**Pull request merged**